client/core/trade.go

```diff
@@ -1007,11 +1013,11 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) boo
 			coinIDString(wallet.AssetID, match.MetaData.Proof.MakerSwap), unbip(wallet.AssetID))
 		confs, spent, err := wallet.SwapConfirmations(ctx, match.MetaData.Proof.MakerSwap,
 			match.MetaData.Proof.ContractData, match.MetaData.Stamp)
-		if err != nil {
-			t.dc.log.Errorf("error getting confirmation for our own swap transaction: %v", err)
+		if err != nil { // unexpected error includes asset.CoinNotFoundError because this is our swap.
+			t.dc.log.Errorf("Unexpected error getting confirmation for our own swap transaction: %v", err)
```

Comment:
OK, this isn't working as expected.  The BTC SPV wallet cannot find _it's own swap txn_, unlike dcrwallet that does.

```
2022-01-15 02:44:33.381 [TRC] CORE: Checking confirmations on our OWN swap txn b61818b246758e7c745047200974f78efbdcbefe790dcc0aec8ebdbd907b7f1a:0 (btc)...
2022-01-15 02:44:33.381 [TRC] CORE[btc][SPV]: swapConfirmations - SpentStatusUnknown: b61818b246758e7c745047200974f78efbdcbefe790dcc0aec8ebdbd907b7f1a:0 (block <nil>, confs 0)
2022-01-15 02:44:33.381 [TRC] CORE[btc][SPV]: swapConfirmations - scanFilters: b61818b246758e7c745047200974f78efbdcbefe790dcc0aec8ebdbd907b7f1a:0 (block <nil>, start time 2022-01-15 02:43:10 +0000 UTC)
2022-01-15 02:44:33.381 [DBG] CORE[btc][SPV]: Performing cfilters scan for b61818b246758e7c745047200974f78efbdcbefe790dcc0aec8ebdbd907b7f1a:0 from height 2135929
2022-01-15 02:44:33.381 [TRC] CORE[btc][SPV]: swapConfirmations - scanFilters did not find b61818b246758e7c745047200974f78efbdcbefe790dcc0aec8ebdbd907b7f1a:0, assuming in mempool.
2022-01-15 02:44:33.381 [ERR] CORE: Unexpected error getting confirmation for our own swap transaction: coin not found
```

Unless an alternate working solution goes up that does not modify huge amounts of code and APIs appears, I'm merging https://github.com/decred/dcrdex/pull/1403